src/backends/tensorrt/tensorrtlib.cc

```diff
@@ -438,17 +463,16 @@ namespace dd
             = predict_dto->parameters->mllib->extract_layer->std_str();
       }
 
-    // detect architecture
-    cudaDeviceProp prop;
-    cudaGetDeviceProperties(&prop, _gpuid);
-    std::string arch = std::to_string(prop.major) + std::to_string(prop.minor);
-    if (_first_predict)
-      this->_logger->info("GPU {} architecture = compute_{}", _gpuid, arch);
-
     TInputConnectorStrategy inputc(this->_inputc);
 
     if (!_TRTContextReady)
       {
+        // detect architecture
+        cudaDeviceProp prop;
+        cudaGetDeviceProperties(&prop, _gpuid);
```

Comment:
This may need to remain outside the conditional, otherwise are we still have to change the GPU id on an existing trt service ?
What's the performance impact ?

Reply:
It takes 0.3ms to query the architecture

The architecture is not supposed to change once we compiled the TRT engine, otherwise we would need to recompile the model anyway, and thus this code would be called again.

I don't think it's possible to change gpuid on one service if the gpu architectures are different, for the same reason: you would have to change the TRT engine, and I don't see any code doing that.

Reply:
OK, thanks, good point. We should be able to change the `gpuid` of a service when architecture is unchanged. Whether we should recompile on the fly whenever the architecture changes is outside the scope of this PR.